src/v2/providers/storage.ts

```diff
@@ -0,0 +1,379 @@
+): CloudFunction<StorageObjectData> {
+  return _onOperation(deletedEvent, buketOrOptsOrHandler, handler);
+}
+
+/** Handle a storage object metadata updated */
+export function onObjectMetadataUpdated(
+  handler: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData>;
+
+export function onObjectMetadataUpdated(
+  bucket: string,
+  handler: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData>;
+
+export function onObjectMetadataUpdated(
+  opts: StorageOptions,
+  handler: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData>;
+
+export function onObjectMetadataUpdated(
+  buketOrOptsOrHandler:
+    | string
+    | StorageOptions
+    | ((event: CloudEvent<StorageObjectData>) => any | Promise<any>),
+  handler?: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData> {
+  return _onOperation(metadataUpdatedEvent, buketOrOptsOrHandler, handler);
+}
+
+/** @hidden */
+export function _onOperation(
+  eventType: string,
+  bucketOrOptsOrHandler:
+    | string
+    | StorageOptions
+    | ((event: CloudEvent<StorageObjectData>) => any | Promise<any>),
+  handler: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData> {
+  if (typeof bucketOrOptsOrHandler === 'function') {
+    handler = bucketOrOptsOrHandler as (
+      event: CloudEvent<StorageObjectData>
+    ) => any | Promise<any>;
+    bucketOrOptsOrHandler = {};
+  }
+
+  const [opts, bucket] = _getOptsAndBucket(
+    bucketOrOptsOrHandler as string | StorageOptions
+  );
+
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(raw as CloudEvent<StorageObjectData>);
+  };
+
+  func.run = handler;
+
+  // TypeScript doesn't recongize defineProperty as adding a property and complains
+  // that __trigger doesn't exist. We can either cast to any and lose all type safety
+  // or we can just assign a meaningless value before calling defineProperty.
+  func.__trigger = 'silence the transpiler';
+
+  Object.defineProperty(func, '__trigger', {
+    get: () => {
+      const baseOpts = options.optionsToTriggerAnnotations(
+        options.getGlobalOptions()
+      );
+      const specificOpts = options.optionsToTriggerAnnotations(opts);
+
+      return {
+        // TODO(colerogers): Remove "apiVersion" once the CLI has migrated to
+        // "platform"
+        apiVersion: 2,
+        platform: 'gcfv2',
+        ...baseOpts,
+        ...specificOpts,
+        labels: {
+          ...baseOpts?.labels,
+          ...specificOpts?.labels,
+        },
+        eventTrigger: {
+          eventType: eventType,
+          resource: bucket, // TODO(colerogers): replace with bucket eventually
+          service: 'storage.googleapis.com',
+        },
+      };
+    },
+  });
+
+  return func;
+}
+
+/** @hidden */
+export function _getOptsAndBucket(
```

Comment:
While you'll still find `_` prefixes in old codebases, that was pre access-control. Also, I think you can use @internal instead of @hidden (would need to double check that it doesn't show up in docs).

Reply:
Removed the `_` prefixes and changed to using `@internal`. The `typedoc` docs indicate that both hidden and internal will remove from the code from documentation as long as we don't pass in `--excludeInternal`.